amethyst_network/src/bundle.rs

```diff
@@ -0,0 +1,95 @@
+extern crate rand;
+
+use super::{ConnectionManagerSystem, NetConnectionPool, NetIdentity, NetReceiveBuffer,
+            NetSendBuffer, NetSocketSystem};
+use amethyst_core::bundle::{ECSBundle, Result};
+use filter::NetFilter;
+use rand::Rng;
+use serde::Serialize;
+use serde::de::DeserializeOwned;
+use shred::DispatcherBuilder;
+use specs::World;
+use std::net::SocketAddr;
+use uuid::Uuid;
+
+/// A convenience bundle to create the infrastructure needed to send and receive network messages.
+pub struct NetworkBundle<'a, T> {
+    /// The local ip to bind to.
+    ip: &'a str,
+    /// The local port to bind to.
+    port: Option<u16>,
+    /// The filters applied on received network events.
+    filters: Vec<Box<NetFilter<T>>>,
+    /// Indicates if this should behaves as a server or as a client when handling remote connections.
+    is_server: bool,
+    /// The server to automatically connect to.
+    /// You would usually want this if you set is_server = false.
+    connect_to: Option<SocketAddr>,
+}
+
+impl<'a, T> NetworkBundle<'a, T> {
```

Comment:
I would probably create two constructors here, one for client, one for server. Makes the distinction a bit clearer.

Reply:
Is this purely just for the network level of messages, or is it slightly higher level as well (I always used a 256-channel API, where each packet could also have different reliability and ordering semantics).